src/Npgsql/Replication/Logical/Protocol/TupleData.cs

```diff
@@ -0,0 +1,63 @@
+﻿using JetBrains.Annotations;
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Npgsql.Replication.Logical.Protocol
+{
+    /// <summary>
+    /// Represents the data transmitted for a tuple in a Logical Replication Protocol message
+    /// </summary>
+    public interface ITupleData
+    {
+        /// <summary>
+        /// The kind of data in the tuple
+        /// </summary>
+        TupleDataKind Kind { get; }
+
+        /// <summary>
+        /// The <see cref="System.Type"/> of <see cref="Value"/>.
+        /// </summary>
+        Type Type { get; }
+
+        /// <summary>
+        /// The value of the tuple, in text format if <see cref="Kind" /> is <see cref="TupleDataKind.TextValue"/>
+        /// or in binary format if <see cref="Kind" /> is <see cref="TupleDataKind.BinaryValue"/>.
```

Comment:
Am not seeing binary [in the docs](https://www.postgresql.org/docs/current/protocol-logicalrep-message-formats.html), this is new and undocumented or something?

Reply:
Ah yes, PG14 :)

Reply:
> Ah yes, PG14 :)

Exactly. I prepared some things already, where I thought it wouldn't hurt.